Add support for signing Secure Boot artifacts using AWS KMS

truls opened the pull request.

truls committed:
Add sbsign patch to allow signing using dynamic engines

truls committed:
Add sbsign patch to allow signing using dynamic engines

truls committed:
Add libsign patches for memory corrpution and engine signing

truls committed:
Add libsign patches for memory corrpution and engine signing

truls committed:
Fix shellcheck linting errors

truls committed:
Fix shellcheck linting errors

truls committed:
Initial support for generating a PKCS11-backed keystore

truls committed:
Initial support for generating a PKCS11-backed keystore

truls committed:
Added recipe for the KMS PKSC11 signing module

truls committed:
Added recipe for the KMS PKSC11 signing module

truls committed:
Prevent additional CFLAG from getting overridden

truls committed:
Prevent additional CFLAG from getting overridden

truls committed:
Add support for signing secure boot artifacts using AWS KMS

truls committed:
Add support for generating config files with yocto-relative dirs

truls committed:
Add support for generating config files with yocto-relative dirs

truls committed:
Generalize path subsitution mechanism and add docs

truls committed:
Generalize path subsitution mechanism and add docs

truls committed:
Always return a dict (prevent variying return types)

truls committed:
Always return a dict (prevent variying return types)

truls committed:
Return correct key path for file-backed signing

truls committed:
Return correct key path for file-backed signing

truls committed:
More documentation

truls committed:
More documentation

tanguy-s commented on meta-signing-key/recipes-devtools/aws-kms-pkcs11/aws-kms-pkcs11_0.0.15.bb:
```diff
@@ -0,0 +1,39 @@
+SUMMARY = "A PKCS#11 interface for AES KMS"
```
```suggestion
SUMMARY = "A PKCS#11 interface for AWS KMS"
```

truls committed:
Also use AWS_SESSION_TOKEN for authentication

truls committed:
Also use AWS_SESSION_TOKEN for authentication

truls committed:
Add support for certificate types for IMA and modsign

truls committed:
Add support for certificate types for IMA and modsign

truls committed:
Add support for signing kernel modules using PKCS#11

truls committed:
Add support for signing kernel modules using PKCS#11

truls committed:
Support new AWS C++ SDK

truls committed:
Support new AWS C++ SDK

truls committed:
Add support for aquiring AWS signing creds using a wrapper command

truls committed:
Add support for aquiring AWS signing creds using a wrapper command

truls committed:
Restore signing using file-backed keystore

truls committed:
Restore signing using file-backed keystore

Commit:
grub-efi: refresh patches for grub-2.14

Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

truls committed:
Add scarthgap backwards-compatbility

truls committed:
Add scarthgap backwards-compatbility

frazer-arm committed:
secure-core-common: Move wic files

Since 5bbe4ba0768 in oe-core, the wic search path has been edited from
wic to files/wic. Move files to match.

Signed-off-by: Frazer Carsley <frazer.carsley@arm.com>

chenxy1988 committed:
grub-efi bbappend: update search parameter

grub would report an error message in boot stage as below:

   "error: no such device: ((hd0,gpt1)/EFI/BOOT)/EFI/BOOT/grub.cfg"

Consequently, the root variable is not set, and the intended protection
against cross-device configuration loading (the purpose of the original 2014 commit)
is lost.

The most robust fix is to use the --hint parameter.
This separates the search target from the device hint, avoiding
fragile string concatenation and supporting both prefixed and
non-prefixed $cmdpath formats.

The modification line up with openembedded-core.git
commit:42b530581f724 ("grub: update search parameter")

Signed-off-by: Xiangyu Chen <xiangyu.chen@windriver.com>

yizhao1 committed:
Merge pull request #131 from mferland/mfe/fix_sbat_section_alignment

seloader: align .sbat section to page boundary to avoid shim warning

yizhao1 committed:
Merge pull request #134 from yizhao1/grub

grub-efi: refresh patches for grub-2.14

yizhao1 committed:
Merge pull request #132 from frazer-arm/move-wic-files

secure-core-common: Move wic files

yizhao1 committed:
Merge pull request #133 from chenxy1988/master

grub-efi bbappend: update search parameter

yizhao1 committed:
grub-efi: disable setting page permissions for loaded modules

GRUB 2.14 introduces memory attribute get/set APIs for NX support
and enforces page permissions on loaded modules [1][2]. SELoader
currently does not support setting memory attributes for NX feature,
which causes boot failure.

Disable the feature as a workaround when SELoader is enabled.

[1] https://gitlab.freedesktop.org/gnu-grub/grub/-/commit/09ca66673ad228c736b3d46f31201065373cf866
[2] https://gitlab.freedesktop.org/gnu-grub/grub/-/commit/94649c02670dae12c188cae471b9ae246b48f428

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

yizhao1 committed:
layer.conf: enable GRUB_SIGN_VERIFY by default instead of UEFI_SELOADER

With the updates of shim to 16.1 and GRUB to 2.14, SELoader faces two
issues:

1. Shim 16.1 implements its own LoadImage() function[1], replacing the
implementation provided by UEFI. However, Shim's LoadImage() function
does not implement the Security2 file authentication functionality
required by SELoader.

2. GRUB 2.14 introduces memory attribute get/set APIs for NX support and
enforces page permissions on loaded modules[2][3]. SELoader currently
does not support setting memory attributes for NX features.

Consequently, to ensure the proper functioning of SELoader, we have had
to adopt two workarounds[4][5]. As a result, we no longer recommend
using SELoader. By default, a GPG key is now used to protect and verify
the files utilized by GRUB.

[1] https://github.com/rhboot/shim/commit/bb114a3b92a96875dc71e5e4925bedba5c02f958
[2] https://gitlab.freedesktop.org/gnu-grub/grub/-/commit/09ca66673ad228c736b3d46f31201065373cf866
[3] https://gitlab.freedesktop.org/gnu-grub/grub/-/commit/94649c02670dae12c188cae471b9ae246b48f428
[4] https://github.com/Wind-River/meta-secure-core/commit/d26de315f8b6d7da8d8e84502da45758108cb0a2
[5] https://github.com/Wind-River/meta-secure-core/commit/190ac40d0b23a59408e17500a865508561f3341c

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>

yizhao1 committed:
Merge pull request #135 from yizhao1/seloader-fix

grub-efi: disable setting page permissions for loaded modules

yizhao1 committed:
Merge pull request #136 from yizhao1/fix

layer.conf: enable GRUB_SIGN_VERIFY by default instead of UEFI_SELOADER

BinCaoWR committed:
efitools: fix build error when efi-secure-boot is not in DISTRO_FEATURES

Commit 6d83fbf ("layer.conf: enable GRUB_SIGN_VERIFY by default instead
of UEFI_SELOADER") sets GRUB_SIGN_VERIFY to 1 by default. When
efi-secure-boot is not in DISTRO_FEATURES, UEFI_SB is 0, so
do_install:append() in efitools_1.9.2.bb does not copy LockDown.efi to
${EFI_BOOT_PATH}. However, do_sign:class-target() in efitools.inc still
attempts to sign LockDown.efi because GRUB_SIGN_VERIFY is 1, resulting
in a build error:

  ERROR: Failed to sign: .../efitools/1.9.2/image/boot/efi/EFI/BOOT/LockDown.efi

Additionally, the check_boot_public_key prefunc may fail on build hosts
where gpg is not available:

  ERROR: Failed to import gpg key (...): /bin/sh: 1: --batch: not found

Fix both issues by adding a UEFI_SB check to do_sign:class-target() and
its prefunc condition, so that signing is only attempted when
efi-secure-boot is actually enabled.

Tested the followings build pass with or without efi-secure-boot
in DISTRO_FEATURES:

  bitbake efitools
  bitbake linux-yocto

Signed-off-by: Bin Cao <bin.cao.cn@windriver.com>

yizhao1 committed:
Merge pull request #137 from BinCaoWR/master

efitools: fix build error when efi-secure-boot is not in DISTRO_FEATURES

Commit:
key-store: Fix incorrect Python membership test syntax

The original expression `not (uks_signing_model(d) in "sample", "user")`
is parsed as `not ((uks_signing_model(d) in "sample"), "user")` due to
Python operator precedence. The `in` operator binds tighter than the comma,
so this creates a tuple where the second element "user" is always truthy,
making the `not` always evaluate to False and the function always returns
early.

Change to `not uks_signing_model(d) in ("sample", "user")` which correctly
tests whether the return value is a member of the tuple ("sample", "user").

Signed-off-by: Xiaozhan Li <xiaozhan.li.cn@windriver.com>

yizhao1 committed:
Merge pull request #138 from lxz0503/xiaozhan-remote

key-store: Fix incorrect Python membership test syntax

truls committed:
Revert "grub-efi: refresh patch for 2.14"

This reverts commit 732859d3e7f73cede71530d4d4343f768d22bb26.

truls committed:
Revert "grub-efi: refresh patch for 2.14"

This reverts commit 732859d3e7f73cede71530d4d4343f768d22bb26.

ericmeyers12-arthrex committed:
ovmf: update for upstream FIXED_GCCVER -> TARGET_TOOLS rename

The openembedded-core ovmf recipe was updated to replace the
FIXED_GCCVER variable with TARGET_TOOLS. Update the secure boot
append to use the new variable name.

This fixes the following build error:
  error 1005: Not supported target [tpm_enable=false]

The error occurred because the stale FIXED_GCCVER reference caused
the build arguments to be misinterpreted by the EDK2 build system.

See upstream openembedded-core commit 325c4b0b14731b4c1782277f49b972e2b1b55d7d
for the original recipe changes.

Signed-off-by: Eric Meyers <eric.meyers@arthrex.com>

yizhao1 committed:
Merge pull request #140 from ericmeyers12-arthrex/master

ovmf: update for upstream FIXED_GCCVER -> TARGET_TOOLS rename

truls committed:
fix: add retry mechanism to signing module to handle aws timeouts

truls committed:
Add support for signing secure boot artifacts using AWS KMS
